### DISCUSSION TITLE
发送20kb的图片却报错图片文件过大

### DISCUSSION THREAD

**Discussion by TDK1969** [nonebot/discussions#112] [问答] [answered]
# 运行环境：

- ubuntu20.04
- nonebot2
- go-cqhttp：v1.0.0-rc1

# 问题描述
我想通过私聊发送图片，但是发现无论是使用url发送还是下载到本地再发送，都会报错如下：

> [2022-02-25 11:41:40] [WARNING]: 警告: 私聊 ********* 消息图片上传失败: file size over max

相关代码如下：
```python3
imgSend = on_command("imgSend", rule=to_me, aliases={"发图", "图片测试"})
@imgSend.handle()
async def AddBiliVideoFollow(matcher: Matcher, args: Message = CommandArg()):
    imgPath = '/root/project/NoneBot/TDK_Bot/src/file/biliStream/covers/haya.jpg'
    imgURI = pathlib.Path(imgPath).as_uri()
    size = os.path.getsize(imgPath) / 1024
    textMsg = '图片大小: {} kb'.format(size)    
    reportedMsg = Message([
        MessageSegment(type='text', data={'text':textMsg}),
        MessageSegment(type='image', data={'url':imgURI})
    ])
    await imgSend.finish(reportedMsg)
```
在私聊中收到消息如下：
![image](https://user-images.githubusercontent.com/54183084/155662997-aa420574-9e65-47e8-bd21-1f43269afb5b.png)

仅收到文本，但没有收到图片

请问我该如何解决这个问题，诚心求助QAQ

**Comment by yanyongyu** [comment, marked as answer]
请正确使用`MessageSegment`，你的image data填写错误，应该是file而不是url

```python
await imgSend.finish(textMsg + MessageSegment.image(Path(imgPath)))
```

**Reply by TDK1969** [reply]
多谢，已解决
但是我是参照nonebot的网站的[这里](https://v2.nonebot.dev/docs/tutorial/process-message#nonebot2-%E4%B8%AD%E7%9A%84%E6%B6%88%E6%81%AF)做的
请问问题具体出在哪里呢？
以及我使用的适配器是onebot.v11

**Reply by yanyongyu** [reply]
文档里的仅作为样例，具体的数据格式由adapter自行定义

**Reply by TDK1969** [reply]
好的，十分感谢！
